libpod/container.go

```diff
@@ -447,6 +449,19 @@ type ContainerNamedVolume struct {
 	Options []string `json:"options,omitempty"`
 }
 
+// ContainerOverlayVolume is a overlay volume that will be mounted into the
+// container. Each volume is a libpod Volume present in the state.
+type ContainerOverlayVolume struct {
+	// Destination is the absolute path where the mount will be placed in the container.
+	Dest string `json:"dest"`
+	// Type specifies the mount kind.
+	Type string `json:"type,omitempty" platform:"linux,solaris"`
+	// Source specifies the source path of the mount.
+	Source string `json:"source,omitempty"`
```

Comment:
Can't we have multiple sources in an overlay mount?

Reply:
Can multiple sources be used by allowing multiple `-v source:dest` with same dest?

Reply:
I remember we had a discussion about this. @rhatdan Did we decide to allow multiple sources? If there's any possibility, we should make this an array.

Reply:
If we want that, we probably need a different command, maybe something with --mount.  I think for --volume we should keep it simple and not complicate the option parsing to much.

Reply:
Change back to treat multiple overlay sources with the same dest as conflict